docs/Documentation.md

```diff
@@ -131,11 +132,163 @@ add semicolons at the end of lines e.g.
+
+```fsharp
+let value = MyFunction()
+let value = person.ToString()
+```
+
+becomes
+
+```fsharp
+let value = MyFunction ()
+let value = person.ToString ()
+```
+
+###### `"SpaceBeforeUnitArgumentInLowercaseInvocation"`
+
+Will add a whitespace before `()` argument when the function name starts with a lowercase character. e.g.
+
+> default false
+
+```fsharp
+let value = myFunction()
+```
+
+becomes
+
+```fsharp
+let value = myFunction ()
+```
+
+###### `"SpaceBeforeParenthesesInUppercaseInvocation"`
+
+> default false
+
+Will add a whitespace before an argument wrapped with parentheses when the function name starts with an uppercase character. e.g.
+
+```fsharp
+let value = MyFunction(a+b)
+```
+
+becomes
+
+```fsharp
+let value = MyFunction (a + b)
+```
+
+###### `"SpaceBeforeParenthesesInLowercaseInvocation"`
+
+> default true
+
+Will add a whitespace before an argument wrapped with parentheses when the function name starts with an lowercase character. e.g.
+
+```fsharp
+let value = myFunction(a+b)
+```
+
+becomes
+
+```fsharp
+let value = myFunction (a+b)
+```
+
+To remove the space, set `"SpaceBeforeParenthesesInLowercaseInvocation"` to `false`.
+
+###### `"SpaceBeforeUnitParameterInUppercaseFunctionDefinition"`
+
+> default false
+
+Will add a whitespace before a `()` parameter in a function definition that starts with an uppercase letter. e.g.
 
-if being set, no space is inserted before a function name and its first argument. 
-For example, `Seq.filter (fun x -> x > 2)` becomes `Seq.filter(fun x -> x > 2)`. 
-This doesn't affect methods and constructors, e.g. `Console.WriteLine("Hello World")`.
+```fsharp
+let Value() = x
+``` 
+
+becomes
+
+```fsharp
+let Value () = x
+```
+
+###### `"SpaceBeforeUnitParameterInLowercaseFunctionDefinition"`
+
+> default false
+
+Will add a whitespace before a `()` parameter in a function definition that starts with an lowercase letter. e.g.
+
+```fsharp
+let value() = x
+```
+
+becomes
+
+```fsharp
+let value () = x
+```
+
+###### `"SpaceBeforeParenthesesInUppercaseFunctionDefinition"`
+
+> default false
+
+Will add a whitespace before a parameters wrapped with parentheses that starts with an uppercase letter. e.g.
+
+```fsharp
+let Value(a:int) = x
+```
+
+becomes
+
+```fsharp
+let Value (a:int) = x
+```
+
+###### `SpaceBeforeParenthesesInLowercaseFunctionDefinition"`
+
+> default true
+
+Will add a whitespace before a parameters wrapped with parentheses that starts with an uppercase letter. e.g.
+
+```fsharp
+let value(a:int) = x
+```
+
+becomes
+
+```fsharp
+let value (a:int) = x
+```
+
+To remove the space, set `"SpaceBeforeParenthesesInLowercaseFunctionDefinition"` to `false`.
+
+###### `"SpaceBeforeClassConstructor"`
+
+> default false
+
+Will add a whitespace before a constructor of class definition.
+
+```fsharp
+type Person() =
+    class end
+```
+
+becomes
+
+```fsharp
+type Person () =
+    class
+    end
+```
```

Comment:
need update

Reply:
I would update all documentation in #725 as the settings do change often when working on these PRs.

Reply:
good idea